pkg/config/teeconfig/teeconfig.go

```diff
@@ -156,11 +164,9 @@ func (t *teeConfig) AllKeysLowercased() []string {
 	return base
 }
 
-func compareResult(key, method string, base, compare interface{}) interface{} {
+func (t *teeConfig) compareResult(key, method string, base, compare interface{}) interface{} {
 	if !reflect.DeepEqual(base, compare) {
-		_, file, line, _ := runtime.Caller(2)
-		fileParts := strings.Split(file, "DataDog/datadog-agent/")
-		log.Warnf("difference in config: %s(%s) -> base: %v | compare %v from %s:%d", method, key, base, compare, fileParts[len(fileParts)-1], line)
+		log.Warnf("difference in config: %s(%s) -> base[%s]: %#v | compare[%s] %#v | from %s", method, key, t.baseline.GetSource(key), base, t.compare.GetSource(key), compare, getLocation(2))
 	}
 	return compare
```

Comment:
This seems like a bug: `compareResult` is returning `compare` which is the nodetreemodel value. It should be returning `base` instead, which is the value from viper.